pkg/alicloud/migration.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package alicloud
+
+import (
+	"encoding/json"
+	api "github.com/gardener/machine-controller-manager-provider-alicloud/pkg/alicloud/apis"
+	"github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
+	"github.com/gardener/machine-controller-manager/pkg/util/provider/machinecodes/codes"
+	"github.com/gardener/machine-controller-manager/pkg/util/provider/machinecodes/status"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	// ProviderAlicloud string const to identify Alicloud provider
+	ProviderAlicloud = "Alicloud"
+)
+
+func migrateMachineClass(alicloudMachineClass *v1alpha1.AlicloudMachineClass, machineClass *v1alpha1.MachineClass) error {
```

Comment:
Suggestion: Adding simple test to verify migration might be helpful to check any breaking changes in future. Refer - https://github.com/gardener/machine-controller-manager-provider-aws/blob/master/pkg/aws/core_test.go#L965-L1120. You can directly test `migrateMachineClass()` if mocking the driver is difficult.

Reply:
agreed, will add unit test here.

Reply:
Thanks.